src/Agents/BaseAgent.php

```diff
@@ -18,6 +18,7 @@ class BaseAgent {
     public $chatModel;
     public int $maxFunctionCalls = 10;  //max number of function loops that can occur without more user input.
     public string $prePrompt = "You are a helpful generalist assistant.";
+    public $lastCallMetadata = null;
```

Comment:
can just be `public $lastCallMetadata`.  Or if we think it will be a nullable array I think we can do 

```
public ?array $lastCallMetadata
```